File: js/src/decoder/FloatDecoder.ts
import { IntWrapper } from './IntWrapper';
import { StreamMetadata } from '../metadata/stream/StreamMetadata';
import { DecodingUtils } from './DecodingUtils';

class FloatDecoder {
    private constructor() {}

    public static decodeFloatStream(data: Uint8Array, offset: IntWrapper, streamMetadata: StreamMetadata): Float32Array {
        const values = DecodingUtils.decodeFloatsLE(data, offset, streamMetadata.numValues());
        const valuesList: Float32Array = new Float32Array(values.length);
        for (let i = 0; i < values.length; i++) {
            valuesList[i] = values[i];
        }
        return valuesList;
    }
}

export { FloatDecoder };

File: java/mlt-core/src/main/java/org/maplibre/mlt/decoder/FloatDecoder.java
package org.maplibre.mlt.decoder;

import java.util.ArrayList;
import java.util.List;
import me.lemire.integercompression.IntWrapper;
import org.maplibre.mlt.metadata.stream.StreamMetadata;

public class FloatDecoder {
  private FloatDecoder() {}

  public static List<Float> decodeFloatStream(
      byte[] data, IntWrapper offset, StreamMetadata streamMetadata) {
    var values = DecodingUtils.decodeFloatsLE(data, offset, streamMetadata.numValues());
    var valuesList = new ArrayList<Float>(values.length);
    for (var value : values) {
      valuesList.add(value);
    }
    return valuesList;
  }
}
